### PR TITLE
Add "make docs" with typedoc for docs/developer/api

### PR DESCRIPTION
**Pull request by illume**
We generate `api` under docs/development.

Using typedoc for the generated `api` files.

# How to use

`make docs`

# Testing done

`make docs` outputs static markdown, and you can view that in the following folders:

- docs/development/api

With the website using the docs branch at (https://github.com/kinvolk/docs/pull/14) inside the website repo you can do `make docs run` and have it serve the api folder with hugo.
- http://localhost:1313/docs/headlamp/0.2/development/api/

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 backend/server
 backend/server.exe
 backend/tools
-app/electron/src/*
\ No newline at end of file
+app/electron/src/*
+docs/development/api-reference/
```
Comment (joaquimrocha) [inline .gitignore:5]: Let's call this simply `docs/development/api`.

**Review by joaquimrocha — COMMENTED** [review]
The API docs should ideally be generated in markdown, maybe this helps: https://www.npmjs.com/package/typedoc-plugin-markdown

Also commented on the kinvolk/docs repo.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
We still need some work in the integration:

Screenshot from 2021-02-25 12-10-20

We need to lose the breadcrumb is generates and add the front-matter to the doc files:
```
---
title: THIS TITLE NEEDS TO BE ADDED
---
```

**Comment by illume** [comment]
Indeed. I have a discussion with the typedoc-plugin-markdown author on how best to proceed generating the front-matter. Will come back to this PR when that discussion is finished.

**Comment by joaquimrocha** [comment]
> Indeed. I have a discussion with the typedoc-plugin-markdown author on how best to proceed generating the front-matter. Will come back to this PR when that discussion is finished.

Great! Thanks.

BTW, regarding Storybook, I am still a bit reluctant that it'll slow down the website build, so let's keep it out and just add insttructions in the docs on how to run it.
I guess running it locally also makes sense if you're looking into how to develop things anyway.

**Comment by illume** [comment]
There's a fancy new theme for Hugo now thanks to the typedoc-plugin-markdown author which adds front-matter. This branch now uses the first release of the theme. It still has some rough edges with links in the side bar and slugs, which I'll debug.

**Comment by illume** [comment]
This now:
- does not build storybook with `make docs`
- looks good in the website
- uses the new hugo theme for the typedoc markdown, with some small fixes to the output until/if that plugin is improved at some point in the future.

The broken test is because of an unrelated issue, see temp hotfix in PR: https://github.com/kinvolk/headlamp/pull/211

**Comment by illume** [comment]
There's a new typedoc-hugo plugin for typedoc markdown that fixes the remaining issues, and I removed the tidy-up script.

**Comment by joaquimrocha** [comment]
I have added an extra option to the typedoc's config that hides the TOC on pages (since we already have one being generated by default).

**Pull request merged**